sklearn/model_selection/_search.py

```diff
@@ -774,15 +777,24 @@ def evaluate_candidates(candidate_params):
         self.cv_results_ = results
         self.n_splits_ = n_splits
 
+        if self.return_all_estimators:
+            self.all_estimators_ = results["estimators"]
+
         return self
 
     def _format_results(self, candidate_params, scorers, n_splits, out):
         n_candidates = len(candidate_params)
 
         # if one choose to see train score, "out" will contain train score info
-        if self.return_train_score:
+        if self.return_train_score and not self.return_all_estimators:
```

Comment:
Hi @jnothman. The changes you've commented about are not in line with my latest commits, please check the changes starting at https://github.com/scikit-learn/scikit-learn/pull/17048/commits/eecf3363762bc872f7204dce107688a519ee2bba. 

In my earlier commits I thought returning the already trained models of each fold was what was being asked for. However, what was truly being asked is to train a model using each parameter set on all the data. I do see a point for this, since many times a data scientist needs to evaluate multiple models on a test set and not interested only in the best fit. What do you think?

Reply:
Apologies, I thought that comment was posted a while ago.